qiskit/ignis/verification/randomized_benchmarking/rb_utils.py

```diff
@@ -245,3 +244,197 @@ def twoQ_clifford_error(ngates, gate_qubit, gate_err):
 
     return (1-alpha2Q_cliff)*3/4
+
+
+def calculate_1q_epg(gate_per_cliff: Dict[int, Dict[str, float]],
+                     epc: float,
+                     qubit: int) -> Dict[str, float]:
+    r"""
+    Convert error per Clifford (EPC) into error per gates (EPGs) of single qubit basis gates.
+    Given that a standard 1Q RB sequences consist of ``u1``, ``u2`` and ``u3`` gates,
+    the EPC can be written using those EPGs:
+
+    .. math::
+
+        EPC = 1 - (1 - EPG_{U1})^{N_{U1}} (1 - EPG_{U2})^{N_{U2}} (1 - EPG_{U3})^{N_{U3}}.
+
+    where :math:`N_{x}` is the number of gate :math:`x` per Clifford.
+    Assuming ``u1`` composed of virtual-Z gate, ie FrameChange instruction,
+    the :math:`EPG_{U1}` is estimated to be zero within the range of quantization error.
+    Therefore the EPC can be written as:
+
+    .. math::
+
+        EPC = 1 - (1 - EPG_{U2})^{N_{U2}} (1 - EPG_{U3})^{N_{U3}}.
+
+    Because ``u2`` and ``u3`` gates are respectively implemented by a single and two
+    half-pi pulses with virtual-Z rotations, we assume :math:`EPG_{U3} = 2EPG_{U2}`.
+    Using this relation in the limit of :math:`EPG_{U2} \ll 1`:
+
+    .. math::
+
+        EPC & = 1 - (1 - EPG_{U2})^{N_{U2}} (1 - 2 EPG_{U2})^{N_{U3}} \\
+            & \simeq EPG_{U2}(N_{U2} + 2 N_{U3}).
+
+    Finally the EPG of each basis gate can be written using EPC and number of gates:
+
+    .. math::
+
+        EPG_{U1} &= 0 \\
+        EPG_{U2} &= EPC / (N_{U2} + 2 N_{U3}) \\
+        EPG_{U3} &= 2 EPC / (N_{U2} + 2 N_{U3})
+
+    To run this function, you first need to run a standard 1Q RB experiment with transpiled
+    ``QuantumCircuit`` and count the number of basis gates composing the RB circuits.
+
+    .. jupyter-execute::
+
+        import pprint
+        import qiskit.ignis.verification.randomized_benchmarking as rb
+
+        # assuming we ran 1Q RB experiment for qubit 0
+        gpc = {0: {'cx': 0, 'u1': 0.13, 'u2': 0.31, 'u3': 0.51}}
+        epc = 1.5e-3
+
+        # calculate 1Q EPGs
+        epgs = rb.rb_utils.calculate_1q_epg(gate_per_cliff=gpc, epc=epc, qubit=0)
+        pprint.pprint(epgs)
+
+    In the example, ``gpc`` can be generated by :func:`gates_per_clifford`.
+    The output of the function ``epgs`` can be used to calculate EPG of CNOT gate
+    in conjugation with 2Q RB results, see :func:`calculate_2q_epg`.
+
+    Note:
+        This function presupposes the basis gate consists
+        of ``u1``, ``u2`` and ``u3``.
+
+    Args:
+        gate_per_cliff: dictionary of gate per Clifford. see :func:`gates_per_clifford`.
+        epc: EPC fit from 1Q RB experiment data.
+        qubit: index of qubit to calculate EPGs.
+
+    Returns:
+        Dictionary of EPGs of single qubit basis gates.
+
+    Raises:
+        KeyError: when ``u2`` or ``u3`` is not found,
+            or specified qubit is not included in the gate count dictionary.
+        ValueError: when ``cx`` gate count is nonzero.
+    """
+    try:
```

Comment:
Rather than a using `try` blocks this would be cleaner to do
```suggestion
    if qubit not in qpc_per_qubit:
        raise QiskitError(msg)
```